### ISSUE TITLE
Explicit 'comparison' with mypy

### ISSUE THREAD

**Issue by disconnect3d** [agronholm/typeguard#50]
Hey,

It might be good to put a `typeguard vs mypy` in readme. I know those are different projects (runtime vs static) but that was my first question to a friend who linked it, so I guess other ppl might ask it too :P.

**Comment by mixmastamyk** [comment]
One other reason I tried  and stuck with typeguard instead of mypy or pydantic is that it is 1/100th the size!  From a cursory view it looks like this is a pure Python lib and the others need lots of stuff built with a compiler.  I haven't looked deep to see if that is true however.

Great library by the way—I looked *high and low* for a function like `check_type()` and this is the first place I found it.  ;-)

**Comment by agronholm** [comment]
Thanks. Typeguard has a long way to go though, and has a tiny fraction of the mindshare that the static checkers have.

**Comment by mixmastamyk** [comment]
Maybe a blurb in the docs could help.  It can always grow in the future if needed.

**Comment by oleks** [comment]
Can it not make sense to use both `typeguard` and `mypy`?

**Comment by agronholm** [comment]
Yes, that is how the majority of the users (as I understand) use typeguard. Typeguard is an additional layer of safeguards.

**Comment by oleks** [comment]
> Yes, that is how the majority of the users (as I understand) use typeguard. Typeguard is an additional layer of safeguards.

Cool, but then why do I get an error like this when trying to `mypy` code that uses `typeguard`:

```
error: Cannot find implementation or library stub for module named "typeguard"
note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```

**Comment by agronholm** [comment]
Currently typeguard itself does not pass mypy checks (due to a number of workarounds) so it does not expose its type information to mypy, hence the error. This is something I want to solve, possibly in the next major release, if it's doable. In the meantime, I suggest you have mypy ignore these errors (a lot of libraries don't have type information available).

**Comment by agronholm** [comment]
As of v3.0.0, typeguard now passes mypy checks in strict mode. It also now contains `py.typed`, marking it as a typed library.
